Sending Mail Function

Kaszana102 opened the pull request:
added function to send mails with attachments.
Simple test also included

Kaszana102 committed:
Sending Mail Function

added function to send mails with attachments.
Simple test also included

Kaszana102 committed:
Merge branch 'main' into SendingMail

BognaLew commented on src/Mail.py:
```diff
@@ -0,0 +1,85 @@
+from typing import List
+
+COMPANY_MAIL = "kubaprojektyinicinnego@gmail.com"
+PASSWORD = "cdcm wrfh nbix unvc"
```
This should be moved to environment variable.

Kaszana102 committed:
Mail fix

Moved mail and password to env varriables for safety.
Changed some styling also

Kaszana102 committed:
SendMail -> None

Lenetis commented on src/mail.py:
```diff
@@ -0,0 +1,87 @@
+            otp['content'] = msg.get_payload()[0].get_payload()
+            otp['attachments'] = msg.get_payload()[1].get_payload()
+            otp['sender'] = msg['from']
+            # otp['attachments'] = ''
```
unnecessary comment?

Lenetis replied:
Also, what does "otp" stand for? Maybe a more descriptive variable name would be better?

Kaszana102 replied:
changed for output
comment deleted

Lenetis commented on src/mail.py:
```diff
@@ -0,0 +1,87 @@
+def ReadMail() -> dict:
+    """
+    function used for automatic testing
+    however it doesn't correctly check attachments
```
"however it doesn't correctly check attachments" - what does that mean?

Kaszana102 replied:
comment modified

Kaszana102 replied:
modified to:
function used for automatic testing.
    for attachments field it returns only single attached file name

Lenetis commented on src/mail.py:
```diff
@@ -0,0 +1,87 @@
+        server.sendmail(COMPANY_MAIL, receiver, msg.as_string())
+
+
+def ReadMail() -> dict:
```
I would suggest using typing.Dict instead of dict. (You already used typing.List above)

Lenetis commented on src/mail.py:
```diff
@@ -0,0 +1,87 @@
+
+    # attach files
+    for f in files or []:
+        with open(f, "rb") as fil:
```
Why "fil" and not "file"?

Lenetis commented on src/mail.py:
```diff
@@ -0,0 +1,87 @@
+
+def SendMail(receiver: str, subject: str, content: str, files: List[str]) -> None:
+    """
+    function which sends mail to appropriate address
```
1. Function names in Python should generally be in snake case (SendMail -> send_mail)
2. I would remove the "function which", "function used for", etc. from all docstrings, because it's obvious that it is a function

Kaszana102 committed:
spelling fixes

Kaszana102 committed:
Merge branch 'main' into SendingMail

Kaszana102 committed:
Filename {basename(f)}

Kaszana102 committed:
Bigger fix

Kaszana102 committed:
Merge branch 'main' into SendingMail
